From 8590893335e1ea5cf260bd2d9d970f233ce8bf84 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 02 Feb 2021 11:18:32 +0200
Subject: [PATCH] [SW-31678] sec: i2c: temporary enabling dbi access before boot

This is a temporary fix, it will be removed soon.

in secured mode after revieving the fit image from lkd, or before
resetting the device from lkd, the iatu access is disabled and
dbi access is enabled before runnig the the fit image.
same functionality is implemented for i2c side as well.

Change-Id: Ie7253a4134669b369e52c63ae70028afe04e57cb
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/module_i2c_slave.c b/preboot/module_i2c_slave.c
index 8ea870c..5bfe950 100644
--- a/preboot/module_i2c_slave.c
+++ b/preboot/module_i2c_slave.c
@@ -140,6 +140,9 @@
 
 void i2cslave_fit_run_image(void)
 {
+	/* TODO: temporary fix, will be removed soon */
+	preboot_disable_iatu_access();
+
 	fit_run_image();
 }
 
